Bikes/Bikes.Tests/BikesTests.cs

```diff
@@ -0,0 +1,144 @@
+    /// </summary>
+    [Fact]
+    public void InformationAboutSportBikes()
+    {
+        var expectedModelIds = new List<int> {2, 5, 8};
+
+        var actualIds = fixture.Bikes
+            .Where(bike => bike.Model.Type == BikeType.Sport)
+            .Select(bike => bike.Id)
+            .ToList();
+
+        Assert.Equal(expectedModelIds, actualIds);
+    }
+
+    /// <summary>
+    /// A test that outputs the top 5 bike models by rental duration
+    /// </summary>
+    [Fact]
+    public void TopFiveModelsRentDurationIds()
+    {
+        var expectedModelIds = new List<int> {10, 1, 2, 5, 3}; 
+
+        var actualIds = fixture.Rents
+            .GroupBy(rent => rent.Bike.Model.Id)
+            .Select(group => new
+            {
+                ModelId = group.Key,
+                Model = fixture.BikeModels.First(m => m.Id == group.Key),
+                TotalDuration = group.Sum(rent => rent.RentalDuration)
+            })
+            .OrderByDescending(x => x.TotalDuration)
+            .Select(x => x.Model.Id)
+            .Take(5)
+            .ToList();
+
+        Assert.Equal(expectedModelIds, actualIds);
+    }
+
+    /// <summary>
+    /// A test that outputs the top 5 bike models in terms of rental income
+    /// </summary>
+    [Fact]
+    public void TopFiveModelsProfit()
+    {
+        var expectedModelIds = new List<int> {10, 5, 2, 1, 3};
+
+        var actualIds = fixture.Rents
+            .GroupBy(rent => rent.Bike.Model.Id)
+            .Select(group => new
+            {
+                ModelId = group.Key,
+                Model = fixture.BikeModels.First(m => m.Id == group.Key),
+                TotalProfit = group.Sum(rent => rent.RentalDuration * rent.Bike.Model.RentPrice)
+            })
+            .OrderByDescending(x => x.TotalProfit)
+            .Select(x => x.Model.Id)
+            .Take(5)
+            .ToList();
+        Assert.Equal(expectedModelIds, actualIds);
+    }
+
+    /// <summary>
+    /// A test that outputs information about the minimum, maximum, and average bike rental time.
+    /// </summary>
+    [Fact]
+    public void MinMaxAvgRentalDuration()
+    {
+        const int expectedMin = 1;
+        const int expectedMax = 5;
+        const double expectedAvg = 2.95;
+
+        var actualMin = fixture.Rents.Min(rent => rent.RentalDuration);
+        var actualMax = fixture.Rents.Max(rent => rent.RentalDuration);
+        var actualAvg = fixture.Rents.Average(rent => rent.RentalDuration);
+
+        Assert.Equal(expectedMin, actualMin);
+        Assert.Equal(expectedMax, actualMax);
+        Assert.Equal(expectedAvg, actualAvg);
+    }
+
+    /// <summary>
+    /// A test that outputs the total rental time of each type of bike
+    /// </summary>
+    [Fact]
+    public void TotalRentalTimeByType() 
+    {
+        var expectedSportRentalTime = 17;
+        var expectedMountainRentaltime = 30;
+        var expectedCityRentaltime = 12;
```

Comment:
В данном тесте сюда отлично пойдет **Theory** вместо **Fact**.
Это позволит вам передавать входные параметры для теста и значительно упростить выборку, так как тут у вас уже не будет группировок. Вы сможете отбирать аренду для конкретного типа и сразу суммировать время ареныд.
```suggestion
    [Theory]
    [InlineData(BikeType.Sport, 17)]
    [InlineData(BikeType.Mountain, 30)]
    [InlineData(BikeType.City, 12)]
    public void TotalRentalTimeByType(BikeType bikeType, int expectedRentalTime) 
    {
```

В тесте вы можете обращаться к этим переменным.